CoreUtilities/src/au/gov/asd/tac/constellation/utilities/svg/SVGData.java

```diff
@@ -0,0 +1,346 @@
+/*
+* Copyright 2010-2023 Australian Signals Directorate
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+ */
+package au.gov.asd.tac.constellation.utilities.svg;
+
+import au.gov.asd.tac.constellation.utilities.text.SeparatorConstants;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Data structure to store SVG information and associations.
+ * Tree-like data structure capable of storing SVG types, attributes and children.
+ * acts as a bare-bones object to be able to store SVG Data for association with minimal impact on memory.
+ * 
+ * @author capricornunicorn123
+ */
+public class SVGData {
+
+    private static final Logger LOGGER = Logger.getLogger(SVGData.class.getName());
+    private final String type;
+    private final Map<String, String> attributes;
+    private final Map<String, SVGData> children;
+    private String content;
+    private SVGData parent;
+
+    public SVGData(final String type, final SVGData parent, final Map<String, String> attributes) {
+        if (type == null){
+            throw new NullPointerException("SVGData elements cannot have a type Null");
+        }
```

Comment:
Throwing null pointers isn't usually a good practice. Might be best in this case to just surround the remaining code in a `(type != null)` and leave at that
Alternatively, perhaps the exception you want to use here is `IllegalArgumentException`

Reply:
I've had a think about what I am trying to achieve here and as an alternate design I will create an enum with a set of valid svg types and enforce that the svg type must be one of these tags. That way it assist future developers and avoids throwing errors.